tests/bluetooth/tester/src/btp_cap.c

```diff
@@ -0,0 +1,1028 @@
+		}
+	}
+
+	start_param.type = BT_CAP_SET_TYPE_AD_HOC;
+	start_param.count = stream_count;
+	start_param.stream_params = stream_params;
+
+	err = bt_cap_initiator_unicast_audio_start(&start_param, u_group->cig);
+	if (err != 0) {
+		LOG_ERR("Failed to start unicast audio: %d", err);
+
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t btp_cap_unicast_audio_update(const void *cmd, uint16_t cmd_len,
+					    void *rsp, uint16_t *rsp_len)
+{
+	int err;
+	const uint8_t *data_ptr;
+	const struct btp_cap_unicast_audio_update_cmd *cp = cmd;
+	struct bt_cap_unicast_audio_update_param stream_params[ARRAY_SIZE(set_members) *
+								     MAX_STREAMS_COUNT];
+
+	LOG_DBG("");
+
+	if (cp->stream_count == 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	data_ptr = cp->update_data;
+	for (size_t i = 0; i < cp->stream_count; i++) {
+		struct unicast_connection *u_conn;
+		struct unicast_stream *u_stream;
+		struct bt_conn *conn;
+		struct btp_cap_unicast_audio_update_data *update_data =
+			(struct btp_cap_unicast_audio_update_data *)data_ptr;
+		struct bt_cap_unicast_audio_update_param *param = &stream_params[i];
+
+		conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &update_data->address);
+		if (!conn) {
+			LOG_ERR("Unknown connection");
+
+			return BTP_STATUS_FAILED;
+		}
+
+		u_conn = tester_bap_unicast_conn_get(bt_conn_index(conn));
+		bt_conn_unref(conn);
+		if (u_conn->end_points_count == 0) {
+			/* Connection not initialized */
+
+			return BTP_STATUS_FAILED;
+		}
+
+		u_stream = tester_bap_unicast_stream_find(u_conn, update_data->ase_id);
+		if (u_stream == NULL) {
+			return BTP_STATUS_FAILED;
+		}
+
+		param->stream = &u_stream->audio_stream.cap_stream;
+		param->meta_len = update_data->metadata_ltvs_len;
+		param->meta = update_data->metadata_ltvs;
+
+		data_ptr = ((uint8_t *)update_data) + param->meta_len +
+			sizeof(struct btp_cap_unicast_audio_update_data);
+	}
+
+	err = bt_cap_initiator_unicast_audio_update(stream_params, cp->stream_count);
+	if (err != 0) {
+		LOG_ERR("Failed to start unicast audio: %d", err);
+
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t btp_cap_unicast_audio_stop(const void *cmd, uint16_t cmd_len,
+					  void *rsp, uint16_t *rsp_len)
+{
+
+	int err;
+	const struct btp_cap_unicast_audio_stop_cmd *cp = cmd;
+	struct unicast_group *group;
+
+	LOG_DBG("");
+
+	group = tester_bap_client_find_unicast_group(cp->cig_id);
+
+	err = bt_cap_initiator_unicast_audio_stop(group->cig);
+	if (err != 0) {
+		LOG_ERR("Failed to start unicast audio: %d", err);
+
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+struct bt_cap_initiator_broadcast_subgroup_param
+	cap_subgroup_params[CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT];
+struct bt_cap_initiator_broadcast_stream_param
+	cap_stream_params[CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT]
+		     [CONFIG_BT_BAP_BROADCAST_SRC_STREAM_COUNT];
+static struct bt_cap_initiator_broadcast_create_param create_param;
```

Comment:
Why are these global and/or `static`?

Reply:
The `create_param` is here by oversight. The `cap_subgroup_params` and `cap_stream_params` are here to allow chaining configuration using the `btp_cap_broadcast_source_setup_stream` and `btp_cap_broadcast_source_setup_subgroup` functions. You can call them multiple times via BTP command, before finally calling `btp_cap_broadcast_source_setup` and creating the local broadcast source.